test/hotspot/jtreg/compiler/lib/ir_framework/IR.java

```diff
@@ -98,6 +98,20 @@
      */
     String[] applyIf() default {};
 
+    /**
+     * Accepts a list of feature pairs where each pair is composed of target feature string followed by a true/false
+     * value where a true value necessities existence of target feature and vice-versa.
+     * IR verifications checks are enforced only if all the specified feature constraints are met.
+     */
+    String[] applyIfTargetFeatureAnd() default {};
+
+     /**
+     * Accepts a list of feature pairs where each pair is composed of target feature string followed by a true/false
+     * value where a true value necessities existence of target feature and vice-versa.
+     * IR verifications checks are enforced if any of the specified feature constraint is met.
+     */
+    String[] applyIfTargetFeatureOr() default {};
```

Comment:
I'm not sure if we should follow the existing scheme to also have at least `applyIfTargetFeature` for a single constraint or not. Back there when I've introduced these constraints I was not happy with writing `applyIfAnd/Or` for many tests where I actually did not care about `AND` and `OR`. I guess you can leave it like that and we can come back to this and maybe clean these things up with [JDK-8280120](https://bugs.openjdk.java.net/browse/JDK-8280120) which wants to introduce another attribute to filter based on the architecture.